locales: Init Files For Weblate

dingyifei opened the pull request:
Prepare the repository for Weblate.

dingyifei committed:
scripts: Upload update_locales.sh Script

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>

dingyifei committed:
locales: Initialize Translation pot/po Files And md Files

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>

dingyifei committed:
locales: Manual Merge of nb_NO

Co-authored-by: Allan Nordhøy <epost@anotheragency.no>

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>
Co-Authored-By: Allan Nordhøy <epost@anotheragency.no>

dingyifei committed:
locales: Manual Merge of es

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>
Co-Authored-By: Adolfo Jayme-Barrientos <fitojb@ubuntu.com>

dingyifei committed:
locales: Manual Merge of zh-hans

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>

dingyifei committed:
locales: Initialize tr(Turkish)

Manual initialization for merging Turkish translations.

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>

dingyifei committed:
locales: Manual Merge of tr

Couldn't find Github account of him

Co-authored-by: udo pton <udopton@gmail.com>

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>

dingyifei commented:
I think we should remove the standard build test from `Klipper-translations` since we don't need to compile anything.

dingyifei committed:
locales: Fix Encoding Issues From Manual Merging

The manual compare and merge process led to some encoding errors in the files.

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>

dingyifei committed:
readme: Include Translation Status

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>

KevinOConnor commented:
Makes sense.  You should have commit access to this repository.  (If not, let me know.)  So, feel free to get everything working the way you think it should be.

After everything is working we can review then, and (if desired) rebase the commits in this repo at that point.

I created the translations branch on this repo just to keep things separate from the "master" branch that holds the klipper code.  There may be better ways to do that.

Thanks,
-Kevin

KevinOConnor commented:
FYI, I also added weblate as a collaborator to this repo (as per https://docs.weblate.org/en/weblate-4.7/vcs.html#hosted-push ) and added a "push" webook to weblate (as per https://docs.weblate.org/en/weblate-4.7/admin/continuous.html#github-setup ).

-Kevin

EDIT: Hopefully this addresses your comments at https://github.com/KevinOConnor/klipper/pull/4226#issuecomment-863827154 - if not, let me know.

dingyifei committed:
locales: Change *.pot Location

It adds readability for the newly generated en output markdown files. Script is modified and simplified to support the new directory.

I reran the update-locales.sh to apply these alterations.

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>

dingyifei committed:
locales: Include Code Blocks in *.po/pot

Add -c parameter for md2po, which enforce the inclusion of code blocks in the po files.

Signed-off-by: Yifei Ding <yifeiding@protonmail.com>
